lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +188,130 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Trims the databases to the block prior to the block the given transaction is in.
-   * @returns The last fully processed block after trimming. `undefined` if all data are deleted after trimming.
+   * A faster version of process transactions that requires access to bitcoin data directory
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const fileReader = new BitcoinFileReader(this.bitcoinDataDirectory!);
+    const blockFiles = fileReader.listBlockFiles().sort();
+    let fileIndex = blockFiles.length - 1;
+
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let previousHash = undefined;
+    let numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    const feeInfos = new Array(numOfBlocksToProcess).fill(undefined);
+
+    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
+    // loop through files and process them until we process all blocks needed
+    while (numOfBlocksToProcess !== 0 && fileIndex >= 0) {
+      const blockFileName = blockFiles[fileIndex];
+      fileIndex--;
+      const blockFile = fileReader.readBlockFile(blockFileName);
+      const blockData = BitcoinRawDataParser.parseRawDataFile(blockFile);
+      // for each block within the current file, add fee info and write to db
+      for (let key in blockData) {
+        const block = blockData[key];
+
+        const indexInFeeInfo = block.height - startingBlock.height;
+        if (indexInFeeInfo >= 0 && indexInFeeInfo < numOfBlocksToProcess) {
+          this.addBlockToFeeInfos(feeInfos, indexInFeeInfo, block);
+          await this.processSidetreeTransactionsInBlock(block);
+        }
+      }
+
+      // look through fee info to see if there are confirmed data
+      // we only need to look from numOfBlocksToProcess because we know the elements after it are confirmed
+      let currentFeeData = feeInfos[numOfBlocksToProcess - 1];
+      while (currentFeeData !== undefined && currentFeeData[currentHash] !== undefined) {
+        // delete all unneeded key value pairs and db records
+        const hashes = Object.keys(currentFeeData);
+        for (let hash of hashes) {
+          if (hash !== currentHash) {
+            delete currentFeeData[hash];
+            await this.transactionStore.removeTransactionByTransactionTimeHash(hash);
+          }
+        }
```

Comment:
Consider moving this into a method, perhaps `removeTransactionsWithBlockchainTimeHashes(hashesToRemove)`. (there is a question about what to do with `delete currentFeeData[hash];`